substrate/network-libp2p/src/network_state.rs

```diff
@@ -175,17 +176,35 @@ impl NetworkState {
 		let node_store = if let Some(ref path) = config.net_config_path {
 			let path = Path::new(path).join(NODES_FILE);
 			if let Ok(node_store) = JsonPeerstore::new(path.clone()) {
-				debug!(target: "sub-libp2p", "Initialized peer store for JSON \
-					file {:?}", path);
+				debug!(target: "sub-libp2p", "Initialized peer store for JSON file {:?}", path);
 				NodeStore::Json(node_store)
 			} else {
-				warn!(target: "sub-libp2p", "Failed to open peer storage {:?} \
-					; peers won't be saved", path);
-				NodeStore::Memory(MemoryPeerstore::empty())
+				warn!(target: "sub-libp2p", "Failed to open peer storage {:?}; peers file will be reset", path);
+				fs::remove_file(&path).expect("Failed deleting peers.json");
+
+				// we check for about 1s if the file was really deleted and move on
+				for _x in 0..200 {
+					if !Path::new(&path).exists() {
+						break;
+					} else {
+						debug!("Waiting for effective deletion of invalid/outdate peers.json");
+						thread::sleep(time::Duration::from_millis(5));
+					}
+				}
+
+				if let Ok(peerstore) = JsonPeerstore::new(path.clone()) {
+					debug!("peers.json reset");
```

Comment:
Note that this also happens if the file doesn't exist in the first place, ie. if the user starts polkadot for the first time, so the message is not totally correct.

Reply:
If the file does not exist in the first place, the first test L179 will kick in and the message you mention L196 will never show up.